lua/entities/gmod_tardis/modules/sh_hads.lua

```diff
@@ -10,7 +10,7 @@ if SERVER then
         return self:SetHADS(on)
     end
 
-    ENT:AddHook("OnTakeDamage", "hads", function(self)
+    function ENT:TryTriggerHADS()
         if self:CallHook("CanTriggerHads") == false then return end
```

Comment:
Can we change up the logic in this function so that it returns true or false whether or not HADS was activated? This will help below